### ISSUE TITLE
Extend i18nDirective to accept translation options

### ISSUE THREAD

**Issue by mazswojejzony** [actimeo/ng2-i18next#4]
This is required to properly handle plural forms, context, etc. 
For example see: http://i18next.com/translate/pluralInterval/

It would be great to be able to use i18nDirective like this:

``` html
<span i18n="translation_key" i18n-options="{postProcess: 'interval', count: someComponentVariable}"></span>
```

**Comment by feloytest** [comment]
Yes, I will need to work on this. ASAP

**Comment by mazswojejzony** [comment]
Any updates? :-)

**Comment by feloy** [comment]
Hi,

I just published a 0.1.1 version. 

The notation for your example would be:

`<span i18n='{"translation_key": {"postProcess": "interval", "count": {{someComponentVariable}}}}'></span>`         

There is still log of debug messages in the console if you need to see what happens.

Is that working for you?

**Comment by marta-tatiana** [comment]
I tried to use the feature and passing options works well. 
However, I observed two problems:
- one is that directive is not updated when passed parameter is updated (in your example, if `{{someComponentVariable}}` changes, change is not reflected to directive).
- second is that when there are multiple usages of directive per component, there is an issue with multiple obsservers (only one observer gets the notification about successful init and only one usage is properly rendered).
  I made some simple fixes I think will improve the situation - @feloy would you mind me submitting a PR?

**Comment by actimeo** [comment]
Hi, 
Sure, feel free to submit a PR with your solution. Thanks!

**Comment by marta-tatiana** [comment]
Hi, I've seen change back to observable in the code - have you observed any problems with Promise implementation?

**Comment by actimeo** [comment]
Yes, the modification introduced a problem with the i18n-directives that are started before the i18n module is initialized. With your modifications, these directives wasn't updated after the i18n module is initialized.

**Comment by marta-tatiana** [comment]
That's true, but why is that a problem? Isn't the point of using Observable with subscribe exactly the same, to delay getting the translation until initialization is done? I hope you don't mind the questions, probably I didn't catch some part of the idea - just trying to understand.
